opm/simulators/wells/WellGroupHelpers.hpp

```diff
@@ -523,12 +570,22 @@ namespace Opm {
         return;
     }
 
-    inline double groupFractionFromInjectionPotentials(const Group& group, const Schedule& schedule, const WellStateFullyImplicitBlackoil& wellState, const int reportStepIdx, const int phasePos) {
+    inline double groupFractionFromInjectionPotentials(const Group& group, const Schedule& schedule, const WellStateFullyImplicitBlackoil& wellState, const PhaseUsage& pu, const int reportStepIdx, const Phase& injectionPhase) {
         double groupTotalGuideRate = 0.0;
         const Group& groupParent = schedule.getGroup(group.parent(), reportStepIdx);
+        int phasePos;
+        if (injectionPhase == Phase::GAS && pu.phase_used[BlackoilPhases::Vapour] )
+            phasePos = pu.phase_pos[ pu.phase_pos[BlackoilPhases::Vapour] ];
+        else if (injectionPhase == Phase::OIL && pu.phase_used[BlackoilPhases::Liquid])
+            phasePos = pu.phase_pos[ pu.phase_pos[BlackoilPhases::Liquid] ];
+        else if (injectionPhase == Phase::WATER && pu.phase_used[BlackoilPhases::Aqua] )
+            phasePos = pu.phase_pos[ pu.phase_pos[BlackoilPhases::Aqua] ];
+        else
+            throw("this should not happen");
```

Comment:
I prefer assert() in general for such things, but it's ok. But what should actually happen if we specified a gas rate for a well in an oil/water case? Do we terminate when building the Schedule object?